internal/cli/template_inspect.go

```diff
@@ -0,0 +1,165 @@
+package cli
+
+import (
+	"strings"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/posener/complete"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateInspectCommand struct {
+	*baseCommand
+
+	flagJson bool
+
+	flagID   string
+	flagName string
+}
+
+func (c *ProjectTemplateInspectCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	name := ""
+	if len(args) == 1 {
+		name = args[0]
+	}
+
+	out, _, err := c.ui.OutputWriters()
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	var tref pb.Ref_ProjectTemplate
+	if name != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: name,
+		}
+	} else if c.flagID != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Id{
+			Id: c.flagID,
+		}
+	} else if c.flagName != "" {
+		tref.Ref = &pb.Ref_ProjectTemplate_Name{
+			Name: c.flagName,
+		}
+	} else {
+		c.ui.Output("Missing project template name or id.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	tr, err := c.project.Client().GetProjectTemplate(ctx, &pb.GetProjectTemplateRequest{
+		ProjectTemplate: &tref,
+	})
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+	template := tr.ProjectTemplate
+
+	if c.flagJson {
+		data, err := protojson.MarshalOptions{
+			Indent: "\t",
+		}.Marshal(template)
+		if err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+
+		c.ui.Output(string(data))
+		return 0
+	}
+
+	table := tablewriter.NewWriter(out)
+	table.SetHeader([]string{
+		"ID",
+		"Name",
+		"Summary",
+		"Terraform Module",
+		"Terraform Module Version",
+		"Tags",
+	})
+	table.SetBorder(false)
+
+	table.Rich([]string{
+		template.Id,
+		template.Name,
+		template.Summary,
+		template.TerraformNocodeModule.Source,
+		template.TerraformNocodeModule.Version,
+		strings.Join(template.Tags, ", "),
+	}, []tablewriter.Colors{
+		{},
+		{},
+		{},
+		{},
+		{},
+	})
+
+	table.Render()
+
+	return 0
+}
+
+func (c *ProjectTemplateInspectCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(sets *flag.Sets) {
+		f := sets.NewSet("Command Options")
+
+		f.BoolVar(&flag.BoolVar{
+			Name:   "json",
+			Target: &c.flagJson,
+			Usage:  "Output project information as JSON.",
+		})
+
+		f.StringVar(&flag.StringVar{
+			Name:    "name",
+			Target:  &c.flagName,
```

Comment:
same comment about flag and arg for `name`

Reply:
Resolved in `8329af09e`.